Structura general.

Banneeddm opened the pull request:
Estructura general del proyecto casi lista.

Banneeddm committed:
Structura general.

maalbera commented on .idea/compiler.xml:
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
agregar al gitignore

mrojasabregu replied:
borrate los archivos .idea, 

no se suben esta carpeta ni los /target

maalbera commented on .idea/.gitignore:
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
```
Editar el gitignore:

HELP.md
target/
!.mvn/wrapper/maven-wrapper.jar
!**/src/main/**/target/
!**/src/test/**/target/

### STS ###
.apt_generated
.classpath
.factorypath
.project
.settings
.springBeans
.sts4-cache

### IntelliJ IDEA ###
.idea
*.iws
*.iml
*.ipr

### NetBeans ###
/nbproject/private/
/nbbuild/
/dist/
/nbdist/
/.nb-gradle/
build/
!**/src/main/**/build/
!**/src/test/**/build/

### VS Code ###
.vscode/

Banneeddm committed:
Estructura Completa.

maalbera commented on spotify-service/src/main/java/com/spotifyservice/spotifyservice/domain/Track.java:
```diff
@@ -1,14 +1,20 @@
-    private int idAlbum;
-    private int reproduction;
-    private int duration;
+    private Long idArtist;
```
Usar Objeto Artist en vez de int.

maalbera commented on spotify-service/src/main/java/com/spotifyservice/spotifyservice/domain/Track.java:
```diff
@@ -1,14 +1,20 @@
+    private Long id;
     private String name;
-    private int idArtist;
-    private int idAlbum;
```
Usar Objeto Album en vez de int

maalbera commented on spotify-service/src/main/java/com/spotifyservice/spotifyservice/configuration/TrackConfiguration.java:
```diff
@@ -0,0 +1,28 @@
+    Track track4 = new Track(4L, "name4", 4L, 4L, 0L, 2.34);
+    Track track5 = new Track(5L, "name5", 5L, 5L, 0L, 5.32);
+
+    @Bean(name = "/{idTrack}")
```
al bean le pusiste no nombre "/{idTrack}"
No esta mal. Pero se presta a confusión, lo podrías tomar a simple vista como una URL

maalbera commented on spotify-service/src/main/java/com/spotifyservice/spotifyservice/domain/Track.java:
```diff
@@ -1,14 +1,20 @@
-    private int id;
+    private Long id;
     private String name;
-    private int idArtist;
```
Usar objeto Artist en vez de int

mrojasabregu commented on spotify-service/.mvn/wrapper/MavenWrapperDownloader.java:
```diff
@@ -0,0 +1,117 @@
+            char[] password = System.getenv("MVNW_PASSWORD").toCharArray();
+            Authenticator.setDefault(new Authenticator() {
+                @Override
+                protected PasswordAuthentication getPasswordAuthentication() {
```
los .mvn tampoco lo subimos

mrojasabregu commented on spotify-service/pom.xml:
```diff
@@ -0,0 +1,68 @@
+	</parent>
+	<groupId>com.spotify-service</groupId>
+	<artifactId>spotify-service</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
```
1.0.0

mrojasabregu commented on spotify-service/pom.xml:
```diff
@@ -0,0 +1,68 @@
+	<artifactId>spotify-service</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+	<name>spotify-service</name>
+	<description>Demo project for Spring Boot - spotify-service</description>
```
cambiar la descripción

mrojasabregu reviewed (changes requested):
porfa agregar en los gitignore 

.idea
.mvn
target/

y borra esos archivos del commit para que no salgan nuevamente

Banneeddm committed:
Correcion .gitignore.

Banneeddm committed:
Update pom.xml

maalbera commented on spotify-service/src/main/java/com/spotifyservice/spotifyservice/service/ArtistService.java:
```diff
@@ -3,40 +3,52 @@
     }
-
     public Artist createArtist(ArtistRequest artistRequest){
         return artistMapper.apply(artistRequest);
```
deberias guardar el artista en una lista o un map, para despues poder hacer la edicion y el borrado

Banneeddm committed:
Update .gitignore

Banneeddm committed:
Cambios estructura.

Consultas:
  Al momento de iniciar el programa, hay alguna forma de "definir" que objetos se van a crear primero? Por ejemplo. Quiero que se cree primero el objeto artista antes que el Album, asi puedo agregarlo al momento en el cual se crea el Album.

Banneeddm committed:
Cambios .gitignore

Banneeddm committed:
Post, Put , Get, Delete --> Track, Album , Artist.
